### ISSUE TITLE
Bulk update part disposition

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#1681]
**Is your feature request related to a problem? Please describe.**
When parts are entered into Arctos we typically assign them the disposition "being processed" so that we can tell that they aren't completely processed into the collection (not in their final storage location or may still require preservation, etc.)

Once the preservation is complete and the part reaches its final storage location, we want to change the disposition to "in collection". For tissues, we would ideally make this update at the time a freezer box is placed in its permanent storage position.

**Describe the solution you'd like**
A simple way to bulk edit part disposition

**Describe alternatives you've considered**
Part bulkloader? It doesn't seem to make changes to disposition.

Put items in a loan and then change from "on loan" to "In collection" - but that puts false loan information in the part history

**Additional context**
It would be great if I could make the change from here http://arctos.database.museum/allContainerLeafNodes.cfm?container_id=14841448 without manually clicking each specimen record, finding the appropriate part and changing the disposition manually, but probably not realistic....

**Priority**
I would like to have this resolved by date: 2018-09-30

**Comment by dustymc** [comment]
<img width="252" alt="screen shot 2018-09-12 at 4 09 43 pm" src="https://user-images.githubusercontent.com/5720791/45457963-4ef05600-b6a6-11e8-8521-a196ba90aec2.png">

should do what you want. You can get there from the "specimens" link at the top of the container leaf nodes page.

**Comment by Jegelewicz** [comment]
Ha! I knew there must be an easy way.....

**Comment by Jegelewicz** [comment]
Well, it doesn't quite work because not all of the parts are in the particular box I am working with. (So if I choose part = liver (frozen), I'll end up with some that shouldn't get changed because the part heart (frozen) happens to be in the box.) Make sense?

**Comment by dustymc** [comment]
>Make sense?

Maybe, and not really.

The immediate issue is that the form works on homogeneous parts, yours aren't. I'm not sure I see how to resolve that from that form - a specimen has 15 liver samples scattered around 15 boxes, you've just got one box that contains one liver sample (and 99 other random things) - right?

I don't see much of a technical problem with adding a disposition-updater from some container form, and the leaf nodes form probably makes the most sense for that - it's the only place that displays only the containers which can't have children (and also happen to be parts). But....

The bigger issue, or potential issue that we've been waiting for someone to bring up for a while now, is that you have multiple types of data doing (about??) the same thing. Disposition is necessary for collections that don't have object tracking, but it just seems like pointless work for potentially-misleading data for collections that have fully embraced object tracking. You (should) KNOW where the thing is because you scanned it there, and that system comes with a whole slew of checks - it's fairly difficult to get it wrong, the history will probably save you even if you find a way to do so, and those data are useful for much more than part location. If you're going to invest in making sure something's right, it should be object tracking. Disposition is just a string that most anyone can set for most any reason, and there are a few ways of doing it for lots of specimens without having any real attachment to the specimen (eg, close a loan in which a couple specimens have not actually been returned to the collection). Trying to maintain "lightweight" disposition data when you have much higher quality data doing the same thing doesn't make much sense to me. We could potentially auto-maintain disposition from the object tracking data, or create a new disposition "see object tracking data" for object which have that, or ???

Or maybe I'm missing something and this is one of those places where reality and ideals just don't line up as nicely as we want them to. If that's the case, the disposition-updater from the leaf nodes form is probably sufficient (and it'll be up to the collections to make sure someone uses it).

@DerekSikes @campmlc @KyndallH @ccicero

**Comment by campmlc** [comment]
I think it is valuable to keep a separate disposition field, for a number
of reasons. It is useful to abe able to record when items are in collection
and and available for loan. Just because something is scanned into a
position doesn't mean we know it is in collection -for example,  we scan
things into object tracking that are  in processing (cryovials into a dewar
in the field, prior to cataloging). I don't have time to go through and
change the disposition of each individual item to in collection once they
are scanned into a freezer box and moved into permanent archive, so a bulk
update tool at the freezer box level would be useful. Also,  items sent on
loan do not change their scanned locations in object tracking - we keep the
location to know where to return the item after the loan is complete - but
the item is not in collecction.
Given that, I support the request for a bulk-update tool for disposition.

**Comment by dustymc** [comment]
>keep a separate disposition field

I'm not proposing to get rid of it, just offering to help reduce some redundancy.

>Just because something is scanned into a position doesn't mean we know it is in collection 

Of course not - it could be in a position in a box which was sent out for loan.

>cryovials into a dewar in the field

There's a great use case. Scan the dewar into the building, and I can do something will all the parts it contains. 

>prior to cataloging

Only cataloged items have parts and only parts have disposition so nothing I can possibly do would have any effect on that.

>I don't have time to go through and change the disposition of each individual item to in collection once they are scanned into a freezer box and moved into permanent archive

That's what I'm offering to help with.

>a bulk update tool at the freezer box level would be useful

I can do that. Using it will of course be left to ya'll.

> items sent on loan do not change their scanned locations in object tracking - we keep the location to know where to return the item after the loan is complete

What I'm proposing will be of no value to you under those procedures.

**Comment by Jegelewicz** [comment]
> you've just got one box that contains one liver sample (and 99 other random things) - right?

Yep.

> Disposition is just a string that most anyone can set for most any reason, and there are a few ways of doing it for lots of specimens without having any real attachment to the specimen (eg, close a loan in which a couple specimens have not actually been returned to the collection). Trying to maintain "lightweight" disposition data when you have much higher quality data doing the same thing doesn't make much sense to me. 

Just because it is here, doesn't mean it is "in collection", thus "being processed".  We use object tracking to help us find stuff, but disposition helps us (and anyone else) understand where the part is in the process.

As for this:
> items sent on loan do not change their scanned locations in object tracking - we keep the location to know where to return the item after the loan is complete

We could probably use some guidance to do this better.  Frozen tissues are a little different in that we remove the vial from storage, cut some tissue for a loan, then return the vial to its original place.  If we scan the vial into a new box during the first step (which would obviously be the preferred way to handle things), we leave an open position in the box where it was originally stored.  It probably isn't going to get filled in by anyone because we don't really have staff to go around filling in empty box positions and if we put the vial back elsewhere, we will still be able to find it. BUT, if we keep returning vials to Cryo or a freezer in new places, we end up with a bunch of empty positions (wasted space) in the original boxes (which we don't have time to go around filling in) so we would really prefer to put vials back where they came from. We can do this using part history, but going part by part through history on a 50 part loan is tedious.  Any ideas for handling this so that we (and when I say we, I'm talking about students) can make the process of returning items taken out for loan sub-sampling to their original positions? Perhaps a loan report that shows the last location of the loan items before they were scanned into the loan box? Ideas appreciated!

**Comment by dustymc** [comment]
>Just because it is here, doesn't mean it is "in collection", thus "being processed". 

That could be as fine-grained as you need - "here" could be in a freezer, or in THAT freezer, or whatever.

>guidance

At least in theory, I always recommend scanning anything you touch. I think that's probably a long-term time saver, and a long scan history usually makes it easy to find lost things if someone does find a way to mess it up. I'm not sure anyone's ever really bought in to that so who knows what the reality looks like.

The tissue sample thing is just weird (and that weirdness is going to be more exposed with the new GenBank submission form) - you (and everybody else) take things out of the object tracking system at the most critical point, when you hand them off to someone who probably isn't familiar with your tracking system, numbering system, might not really get what rigorously documenting specimen usage looks like, etc. I don't see a defensible solution that doesn't involve barcoding subsamples, and I'm not sure how realistic that might be. ("Not very" seems likely....)

I'm not sure if you could fit them on the tiny tubes with any existing encoding, but this would be a really great place to print ARKs as barcodes. With that, citing the barcode (which seems to happen with a fair bit of regularity when there are barcodes) would still resolve to the specimen.

Here's an ARK ( https://n2t.net/ark:/87299/x6td9vc6 ) in MicroPDF417 if you want to play with that idea.

![barcode-2](https://user-images.githubusercontent.com/5720791/45513993-0a71c280-b759-11e8-93f2-2968596a60c6.gif)

We could do a lot more to integrate containers and loans, and a few of us have spent a lot of time thinking about that, I just need someone with a real use-case involving barcoded parts. "Find a box [in {pick freezer}] with {enter a number} of empty slots" could be part of Arctos - that'd be useful for loan returns and anything else that generates tubes. "Print next-to-last scan locations for this returning loan" should be easy enough as well, but isn't quite so versatile.

**Comment by Jegelewicz** [comment]
> The tissue sample thing is just weird (and that weirdness is going to be more exposed with the new GenBank submission form) - you (and everybody else) take things out of the object tracking system at the most critical point, when you hand them off to someone who probably isn't familiar with your tracking system, numbering system, might not really get what rigorously documenting specimen usage looks like, etc. I don't see a defensible solution that doesn't involve barcoding subsamples, and I'm not sure how realistic that might be. ("Not very" seems likely....)

Yes and no - the subsample sent off is still attached to the barcoded part as well as the loan (we don't put the part in the loan, just the subsample). I don't see any reason we couldn't put a barcode on the subsample at the time we were packing the loan and perhaps that is where the ARK comes in? If it would make it easier for a borrower to cite us, I think it would be something we would be interested in...

**Comment by campmlc** [comment]
The problem is not the subsamples. If those are returned, they are put in a
new place. It is the fact that we remove the parent sample from its
location and put it in a temporary location for loan processing, to be
returned eventually. But in the meantime, there is no way to track where
the parent sample actually is. We currently do not scan it into a new
position because we need to know where the old position is to return the
sample.

We could do the ideal thing, e.g. scan all loan items into a new box,
process them, and then return, but only IF we had a way to "see the last
recorded location of all collection objects in this container", e.g. not
their current location, but their previous locations.

**Comment by dustymc** [comment]
>still attached to the barcoded part

Hu?

>make it easier for a borrower to cite us

Harder to avoid anyway - given a tube that says "6" on it (and explicit instructions to cite http://arctos.database.museum/guid/MSB:Mamm:292063) some users will do what you tell them and some will do nothing and some will cite "6" (which is useless). Given the same situation but "https://n2t.net/ark:/87299/x6td9vc6" instead of "6" on the tube, the ones who'd cite whatever's in front of them will inadvertently cite something useful anyway. (And it goes to the part which can be linked to an event - in that case it's better than the GUID.)

>there is no way to track where the parent sample actually is. 

Here's what I'd propose:

* Pull the tissues, scan them into a working box (boxes, cold plate, whatever - something you can stick a barcode on)
* Do your loan thing. If someone else wants to borrow more bits you can find them any time.
* Find some empty box positions - we can do the two things I mentioned and probably more - and scan them back.

**Comment by campmlc** [comment]
This is not trivial:

   - Find some empty box positions - we can do the two things I mentioned
   and probably more - and scan them back.

I want to be able to put them back into the empty positions they were taken
out of. Too difficult to go look for random holes - how exactly? Also, it
minimizes the number of different environmental systems these tubes are
exposed to. Out of freezer 3, back into freezer 3, not into freezer 12 etc.
I want to be able to trust students to do this.

**Comment by Jegelewicz** [comment]
Still attached to the part with the barcode:

![image](https://user-images.githubusercontent.com/5725767/45521367-dad3b200-b77a-11e8-9a76-15ab03030139.png)

**Comment by dustymc** [comment]
How is the tube that was delivered to the loan recipient marked?

The linkage between stuff in Arctos is our problem and resolvable barcodes won't help with that. They might help link results (eg, publications, genbank sequences) to parts.

**Comment by campmlc** [comment]
Not sure how this conversation ended up here. We don't need barcodes to
deal with loan subsamples. We just need a way to bulk update disposition at
the container level. Possible?

a bulk update tool at the freezer box level would be useful

I can do that. Using it will of course be left to ya'll.

**Comment by campmlc** [comment]
Moving loan part location thread to separate issue #1687 .
This issue pending resolution of bulk update tool at freezer location.

**Comment by dustymc** [comment]
added update all option
